ci: master bit-rot — perl-parser --features incremental --tests fails with unresolved import

Every PR smoke run is currently failing **Compile All Targets (bit-rot guard)** because master has a cargo build error when `--features incremental` + `--tests` are combined.

## Reproducer

```bash
cargo check -p perl-parser --features incremental --tests --locked
# OR
cargo check --workspace --all-targets --locked
```

## Error

```
error[E0432]: unresolved import `super::incremental_edit`
    --> crates/perl-parser/src/incremental/incremental_document.rs:1080:16
     |
1080 |     use super::incremental_edit::IncrementalEdit;
     |                ^^^^^^^^^^^^^^^^ could not find `incremental_edit` in `super`
```

## Symptoms

- `cargo check -p perl-parser --features incremental --lib` → passes
- `cargo test -p perl-parser --features incremental --lib --no-run` → fails
- `cargo check --workspace --all-targets --locked` → fails (unified via `perl-lsp-rs/default = ["workspace", "incremental"]`)
- First observed on #5005 after #4988 landed

## Root cause direction

`mod.rs` declares `pub mod incremental_edit;` unconditionally. The identical import works at top-level line 6 but fails inside `#[cfg(test)] mod tests` at line 1080.

## Impact

- PR Smoke / compile-all-targets RED on every PR
- Pre-push hooks fail locally
- Every Codex PR looks broken

## Priority

Blocker. Fix before merging through the merge-gate path.

Route: builder with narrow spec — make `cargo test -p perl-parser --features incremental --no-run` compile.

--- a/crates/perl-parser/src/incremental/incremental_document.rs
+++ b/crates/perl-parser/src/incremental/incremental_document.rs
@@ -1077,7 +1077,7 @@ impl SubtreeCache {
 
 #[cfg(test)]
 mod tests {
-    use super::incremental_edit::IncrementalEdit;
+    use super::super::incremental_edit::IncrementalEdit;
     use super::*;
 
     #[test]

